From 2fb88b055d1d5c92de95388e90a0b4427986f95c Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
Date: Fri, 28 Jan 2022 11:01:57 +0100
Subject: [PATCH] Increase sleep t/integration/reload_syntax_error.t
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

2 seconds were not enough for machines of Fedora CI. The test usually failed
on reloading a damaged module:

    ./t/integration/resource.t ................. ok
	# Failed test 'Exit Value Check'
	# at ./t/integration/reload_syntax_error.t line 91.
	# +-----+----+-------+
	# | GOT | OP | CHECK |
	# +-----+----+-------+
	# | 256 | eq | 0     |
	# +-----+----+-------+
	# Command = /usr/bin/perl /usr/bin/yath -D/usr/share/perl5/vendor_perl run ./t/integration/reload_syntax_error.tx :: fixed
	# Exit = 256
	# ==== Output ====
	# (INTERNAL)     19086 yath-nested-runner-A - Runner detected a change in one or more preloaded modules...
	# (INTERNAL)     19086 yath-nested-runner-A - blacklisting changed files and reloading stage...
	# (INTERNAL)     19103 yath-nested-runner-A - Runner detected a change in one or more preloaded modules...
	# (INTERNAL)     19103 yath-nested-runner-A - blacklisting changed files and reloading stage...
	# (INTERNAL)     Global symbol "$bob" requires explicit package name (did you forget to declare "my $bob"?) at /tmp/rZx4DpKaWI/Preload/Flux.pm line 7.
	# (INTERNAL)     Compilation failed in require at (eval 168) line 1.
	# ( FAILED )  job  1    t/integration/reload_syntax_error.tx
	# < REASON >  job  1    Test script returned error (Signal: 1)
	# < REASON >  job  1    No plan was declared, and no assertions were made.
	#
	# The following jobs failed:
	# +------------------------------------+------------------------------------+
	# | Job ID                             | Test File                          |
	# +------------------------------------+------------------------------------+
	# | 878AB4E0-8017-11EC-A029-6EA139B4B3 | t/integration/reload_syntax_error. |
	# | 6F                                 | tx                                 |
	# +------------------------------------+------------------------------------+
	#
	#                                 Yath Result Summary
	# -----------------------------------------------------------------------------------
	#      Fail Count: 1
	#      File Count: 1
	# Assertion Count: 0
	#       Wall Time: 1.64 seconds
	#        CPU Time: 0.90 seconds (usr: 0.29s | sys: 0.13s | cusr: 0.31s | csys: 0.17s)
	#       CPU Usage: 54%
	#     -->  Result: FAILED  <--
	# [0m
	#
	# ========

    # Failed test 'yath run ./t/integr[...]x_error.tx :: fixed'
    # at ./t/integration/reload_syntax_error.t line 91.
    # Seeded srand with seed '20220128' from local date.
    ./t/integration/reload_syntax_error.t ......
    Dubious, test returned 2 (wstat 512, 0x200)
    Failed 2/5 subtests

Signed-off-by: Petr Písař <ppisar@redhat.com>
---
 t/integration/reload_syntax_error.t | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/t/integration/reload_syntax_error.t b/t/integration/reload_syntax_error.t
index afcb2b4..d445dd8 100644
--- a/t/integration/reload_syntax_error.t
+++ b/t/integration/reload_syntax_error.t
@@ -42,7 +42,7 @@ sub touch {
     my ($inject) = @_;
     my $path = "$tmpdir/Preload/Flux.pm";
     note "Touching $path...";
-    sleep 2;
+    sleep 2*5;
 
     open(my $fh, '>', $path) or die $!;
 
-- 
2.34.1

